receiver/receiverhelper/scraper.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receiverhelper
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// Scraper provides a function to scrape metrics.
+type Scraper interface {
+	Scrape(context.Context) (pdata.Metrics, error)
+}
+
+// Scrape specifies the function that will be invoked every collection interval when
+// the receiver is configured as a Scraper.
+type Scrape func(context.Context) (pdata.Metrics, error)
+
+// ScraperConfig provides functions to get and set the collection interval.
+type ScraperConfig interface {
+	configmodels.Receiver
+
+	CollectionInterval() time.Duration
+	SetCollectionInterval(time.Duration)
+}
+
+// ScraperSettings defines common settings for scraper configuration.
+// Specific scrapers can embed this struct and extend it with more fields if needed.
+type ScraperSettings struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	CollectionIntervalVal         time.Duration `mapstructure:"collection_interval"`
+}
+
+// CollectionInterval gets the scraper collection interval.
+func (ss *ScraperSettings) CollectionInterval() time.Duration {
+	return ss.CollectionIntervalVal
+}
+
+// SetCollectionInterval sets the scraper collection interval.
+func (ss *ScraperSettings) SetCollectionInterval(collectionInterval time.Duration) {
```

Comment:
This feels awfully Java-y. 😛 Maybe can be improved as part of decoupling receiver and scraper as part of the comment above.

Reply:
This is probably overkill, but replicates what the `configmodels.Receiver` interface does so that it can be used in a similar manner (e.g. see usage in `NewScraper` below)